src/App.tsx

```diff
@@ -15,48 +17,131 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortField {
+  name = 'name',
+  length = 'length',
+  empty = '',
+}
+
+type ReorderParams = {
+  sortField: SortField,
+  reverseIsActive: boolean,
+};
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, reverseIsActive }: ReorderParams,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortField.name:
+          return good1.localeCompare(good2);
+
+        case SortField.length:
+          return good1[SortField.length] - good2[SortField.length];
```

Comment:
```suggestion
          return good1.length - good2.length;
```
if someone changes the sortField, the logic will break